frontend/src/advising/Templates/Templates.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+import { withRouter } from "react-router-dom";
+import { NewTemplatesPage } from "./NewTemplatesPage";
+import { TemplatesListPage } from "./TemplateListPage";
+
+export interface TemplateProps {
+  name: string;
+}
+
+interface TemplatesAPI {
+  templates: TemplateProps[];
+  nextPage: number;
+  lastPage: boolean;
+}
+
+interface TemplatesListProps {
+  searchQuery: string;
+}
+
+export enum TemplatePageState {
+  LIST = "LIST",
+  NEW = "NEW",
+}
```

Comment:
This enumeration is used to define each of the pages within the templates flow since they are all under a single URL, currently there is one for showing the list of all templates and one for showing the template creation form.